bucket/openssh.json

```diff
@@ -1,69 +1,49 @@
-                "eda2132707ef438db009613c011b47217cb00ee4d4a28ad4f6cb10ad1845f74e",
-                "333b5089ea0d27c167e7bcf786bf0ceb2192d76c18f338379d771adda18bda3e",
-                "00caef39902d34112fe9f9e5f9aad063cbde3922c212558ce4cec5f8c33534ee",
-                "236bedb6a8a90c0ecb70c1bc9c90ccd347f2f1ae96df11aa9af8e6878a606fe3",
-                "e142b12df8b5f1765a1a0739a5a304d18f810523ac5d36dc66bb92ff00d67d1b",
-                "45dcd0b4ac5444cfa8a7b9b98908b398710b65277e639666135746d7ffe0bfbb",
-                "bc452a4db3ea81ba812b1b153e2486d4abd63d64f4d376b7b7f2133efadd11a5",
-                "0943eb05da94d3ab8efac1a923bfcd515a9ba7251186e29401f06f155ce4d2ba",
-                "bc633859967a1a3f99f97e79daae6a9b520bf2cdc7e7c14e7647f1e86adee289",
-                "4f34fdff9fd67affd4de0c6becf112976df8e77ead0937d86db38cd50839ac3f",
-                "4af63558e39e7a4941292132b2985cb2650e78168ab21157a082613215e4839a"
-            ]
+            "url": "https://github.com/PowerShell/Win32-OpenSSH/releases/download/V8.6.0.0p1-Beta/OpenSSH-Win64.zip",
+            "hash": "9f9215dc0b823264d52603f4767d8531880ddfa9aedf16404923814c0ab086b7",
+            "extract_dir": "OpenSSH-Win64"
+        },
+        "32bit": {
+            "url": "https://github.com/PowerShell/Win32-OpenSSH/releases/download/V8.6.0.0p1-Beta/OpenSSH-Win32.zip",
+            "hash": "0221324212413a6caf260f95e308d22f8c141fc37727b622a6ad50998c46d226",
+            "extract_dir": "OpenSSH-Win32"
         }
     },
-    "pre_install": [
-        "ensure \"$dir\\tmp\" | Out-Null",
-        "ensure \"$dir\\home\" | Out-Null",
-        "ensure \"$dir\\home\\$env:UserName\" | Out-Null",
-        "ensure \"$env:UserProfile\\.ssh\" | Out-Null",
-        "cmd /c mklink /j \"$dir\\home\\%UserName%\\.ssh\" \"%UserProfile%\\.ssh\""
-    ],
     "bin": [
-        [
-            "usr\\bin\\sh.exe",
-            "findssl.sh",
-            "$dir/usr/bin/findssl.sh"
-        ],
-        "usr\\bin\\scp.exe",
-        "usr\\bin\\sftp.exe",
-        "usr\\bin\\ssh.exe",
-        "usr\\bin\\ssh-add.exe",
-        "usr\\bin\\ssh-agent.exe",
-        [
-            "usr\\bin\\sh.exe",
-            "ssh-copy-id",
-            "$dir/usr/bin/ssh-copy-id"
-        ],
-        "usr\\bin\\ssh-keygen.exe",
-        "usr\\bin\\ssh-keyscan.exe",
-        "usr\\bin\\sshd.exe"
+        "scp.exe",
+        "sftp.exe",
+        "sftp-server.exe",
+        "ssh.exe",
+        "ssh-add.exe",
+        "ssh-agent.exe",
+        "ssh-keygen.exe",
+        "ssh-keyscan.exe",
+        "ssh-shellhost.exe",
+        "sshd.exe"
+    ],
+    "checkver": {
+        "github": "https://github.com/PowerShell/Win32-OpenSSH",
+        "regex": "tag/V([\\d.]+p\\d)-Beta"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/PowerShell/Win32-OpenSSH/releases/download/V$version-Beta/OpenSSH-Win64.zip"
```

Comment:
These are beta releases though.

Best of ideas to use this repo?

Reply:
All there releases are beta :)

I guess the non-beta ones are those that ship with the OS updates.

Reply:
That's what I mean

Not the best of ideas in my humble opinion

Reply:
This manifest is not meant for Windows 10 anyway (as mentioned in the note) - since these builds are included with the system. This is for Windows 7, 8, 8.1 users.

Reply:
> as mentioned in the note

It's possible for notes to be ignored

If the note is meant for something like this, then we should display the note while installation starts

With something like

Warning: Not recommended for using with Windows 10

And then ask the user to key in with a Proceed or not. (which ends up breaking scripts)

Other option is to disable installation of this based on Windows versions

We would need to create tooling which finds current Windows versions and checks if installation is allowed based on the Windows versions

Reply:
Clarified the note to reflect this better.

Reply:
Further Git for example ships with OpenSSH 8.8 while this is a beta version of an older OpenSSH release.

Reply:
Currently there is no way in Scoop to check OS versions, and pre-maturely exiting from pre-install scripts would result in a broken installation.

Reply:
We already have a separate manifest for git-with-openssh, users can install that if they want.

Reply:
This is for Windows 7 and up (https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH#install-win32-openssh-test-release)

From 8.1.0.0p1, Microsoft remove `This is a pre-release (non-production ready)` from release note, so I think it is stable enough for production.

Reply:
@niheaven personally I see bundling the beta version of an outdated OpenSSH version (8.6) while the current is 8.8 to be not the best of ideas.

Reply:
Microsoft use OpenSSH 8.6p1 even in Windows 11's latest Insider Preview.